### ISSUE TITLE
Bug in pushing with tuples

### ISSUE THREAD

**Issue by kizu** [stylus/stylus#1066]
There is a bug when we work with multidimensional arrays. When we push an array to an empty array, we get not an array of a 1 length containing the pushed array, but the original pushed array.

This makes things like #677 very hard to do, 'cause we need to make the first contatenation by hand using `$foo = $foo, $bar` for the second `push` in a situation like this:

```
$array = ()
for $i in 1..3
  push($array, ($i $i))

.foo
  val: unquote(join(', ', $array))
```

The code above gives us `val: 1, 1, 2 2,3 3;` while we'd expect it to give us `1 1, 2 2, 3 3`.

The only fix I found is to do so:

```
$array = ()
for $i, $index in 1..3
  $new = ($i $i)
  if $index == 1
    $array = $array, $new
  else
    push($array, $new)

.foo
  val: unquote(join(', ', $array))
```

So, this is obviously a bug. However, fixing it could bring troubles if someone had used this incorrect behaviour in his code.

**Comment by sevenupcan** [comment]
Yes, thanks for the work around. I had been assigning the first item to the array but hadn't thought to assign it to itself and then the first item.

ie:
```
$array = $array, $new
```

**Comment by sevenupcan** [comment]
I think there may still be an issue.

When I try to push into an array multiple times the first expression is surrounded by parentheses. This means when I iterate through the array it counts the first item as several expressions.

```stylus
test = ()

for i in 1..3 {
	push(test, (i i))
}

p(test)

```

which produces

```
inspect: 1 1 (2 2) (3 3)
```

Iterating over the array produces
```
for item in test {
	p(item)
}

inspect: 1
inspect: 1
inspect: 2 2
inspect: 3 3
```

**Comment by sevenupcan** [comment]
So the only workaround that I can come up with with the fix applied is doing the following.

```
array = ""

for i in 1..3 {
    push(array, i i)
}

shift(array)
```

This produces

```
inspect: (1 1) (2 2) (3 3)
```

Maybe someone else might find this useful.

**Comment by sevenupcan** [comment]
Sorry me again, but since working on a project that used the push bif frequently I've come across some issues with the proposed fix. It's difficult to explain so I'll try to demonstrate it with an example.

With the fix it doubles up the parentheses which causes issues when iterating through the list, as you have to iterate through two layers, rather than one.

```
foo = (1 1) (2 2) (3 3)
bar = (a a) (b b) (c c)
foobar = ""

for foz in foo {
	for baz in bar {
		push(foobar, baz)
	}
}

// To remove the first item in the array
shift(foobar)

p(foobar)
```

With the fix produces
```
inspect: ((a a)) ((b b)) ((c c)) ((a a)) ((b b)) ((c c)) ((a a)) ((b b)) ((c c))
```

Without the fix this produces
```
inspect: (a a) (b b) (c c) (a a) (b b) (c c) (a a) (b b) (c c)
```
